functions/index.js

```diff
@@ -0,0 +1,130 @@
+const functions = require('firebase-functions');
+
+// The Firebase Admin SDK to access the Firebase Realtime Database.
+const admin = require('firebase-admin');
+admin.initializeApp(functions.config().firebase);
+
+// // Create and Deploy Your First Cloud Functions
+// // https://firebase.google.com/docs/functions/write-firebase-functions
+
+// THIS FUNCTION ADDS PUSHTOKEN FOR USERID
+exports.addPushTokenForUserId = functions.https.onRequest((req, res) => {
+  const userId = req.query.userId;
+  const pushToken = req.query.pushToken;
+
+  admin.database().ref('/pushTokens/' + userId).set({token: pushToken}).then(snapshot => {
+    console.log('pushToken added for userId ' + userId);
+    res.sendStatus(200);
+  });
+
+});
+
+exports.addNewChatBetweenUsers = functions.https.onRequest((req, res) => {
+  console.log(req.query)
+  const userId1 = req.query.userId1;
+  const userId2 = req.query.userId2;
+  
+  return admin.database().ref('/chats/')
+         .push({
+           messages: [],
+           users: [userId1, userId2]
+         })
+    .then(snapshot => {
+      console.log('chat record was added for users ' + userId1 + ' and ' + userId2);
+      const chatKey = snapshot.ref.toString().split('/').slice(-1)[0];
+      console.log(`chatKey: ${chatKey}`);
+      res.send({ 'chatKey': chatKey });
+      return admin.database().ref(`/chats/${chatKey}/messages`)
+             .push({
+               msg: 'Tervetuloa whappuäpin chättiin!',
```

Comment:
I'd consider adding `env` variable and have `msg` use it as value and default to hard coded string if `env` variable is not set. We wouldn't have to make a back end release to change a single user facing string.